ecommerce/context_processors.py

```diff
@@ -0,0 +1,13 @@
+from django.conf import settings
+from products.models import Category
+from django.core.cache import cache
+
+def categories_processor(request):
+    categories_cache = cache.get(settings.CATEGORIES_CACHE)
+    if categories_cache:
+        categories = categories_cache
+    else:
+        categories = Category.objects.all()
+        cache.set(settings.CATEGORIES_CACHE, categories, 60*60*24)
```

Comment:
It would be nice (but not necessary) to have that type of thing in some config file. Read about django settings if it is appropriate to put it there if no create config file with dotenv support